include/soci/soci-platform.h

```diff
@@ -105,45 +105,24 @@ namespace std {
 // mode, we just need to check for the minimal compiler version supporting them
 // (see https://msdn.microsoft.com/en-us/library/hh567368.aspx).
 
-#if defined(SOCI_HAVE_CXX11) || (defined(_MSC_VER) && _MSC_VER >= 1800)
-# define SOCI_OVERRIDE override
+#ifdef _MSC_VER
+    #if _MSC_VER < 1900
+        #error This version of SOCI requires MSVS 2015 or later.
+    #endif
 #else
-# define SOCI_OVERRIDE
-#endif
-
-namespace soci
-{
-
-namespace cxx_details
-{
-
-#if defined(SOCI_HAVE_CXX11) || (defined(_MSC_VER) && _MSC_VER >= 1800)
-    template <typename T>
-    using auto_ptr = std::unique_ptr<T>;
-#else // std::unique_ptr<> not available
-    using std::auto_ptr;
+    #if __cplusplus < 201402L
+        #error This version of SOCI requires C++14.
+    #endif
```

Comment:
This could be dangerous, as afaik a couple of older compiler versions used to set this macro to garbage and thus hard-erroring if this macro contains an invalid value seems a bit too much. :thinking:

Reply:
I've never seen any compilers setting `__cplusplus` to garbage, and if there really are any, they certainly can't be used for compiling C++11 code. Perhaps you were thinking about MSVC, which traditionally didn't set it correctly, which is why it's handled specifically above.